tests/unit_tests/steps/test_grab.py

```diff
@@ -15,24 +19,37 @@
 
 
 class TestGrabFolder:
+    '''Test grab folder functionality.'''
 
     def test_trailing_slash(self):
-        with pytest.warns(UserWarning, match="_metric_send_conn not set, cannot send metrics"):
-            self._common(grab_src='/grab/source/', expect_grab_src='/grab/source/')
+        '''Test folder grabbing with a trailing slash.'''
+        with pytest.warns(UserWarning, match="_metric_send_conn not set, "
+                                             "cannot send metrics"):
+            self._common(grab_src='/grab/source/',
+                         expect_grab_src='/grab/source/')
 
     def test_no_trailing_slash(self):
-        with pytest.warns(UserWarning, match="_metric_send_conn not set, cannot send metrics"):
-            self._common(grab_src='/grab/source', expect_grab_src='/grab/source/')
+        '''Test folder grabbing without a trailing slash.'''
+        with pytest.warns(UserWarning, match="_metric_send_conn not set, "
+                                             "cannot send metrics"):
+            self._common(grab_src='/grab/source',
+                         expect_grab_src='/grab/source/')
 
     def _common(self, grab_src, expect_grab_src):
         source_root = Path('/workspace/source')
         dst = 'bar'
 
         mock_config = SimpleNamespace(source_root=source_root,
                                       tool_box=ToolBox())
+        # Since is_available calls run, in order to test a single run call,
+        # we patch is_available to be always true.
         with mock.patch('pathlib.Path.mkdir'):
             with mock.patch('fab.tools.tool.Tool.run') as mock_run:
-                grab_folder(mock_config, src=grab_src, dst_label=dst)
+                with mock.patch(
+                        'fab.tools.tool.Tool.is_available',
+                        new_callable=mock.PropertyMock) as is_available:
+                    is_available.return_value = True
```

Comment:
This may not be necessary if we support only Posix shell.

Reply:
I think we should still try if even `sh` is available - who knows ;) And the advantage is that we can then use `assert_called_once_with` (instead of having to check for two calls).